shell/monitor/monitor.sh

```diff
@@ -34,9 +39,71 @@ _monitor_set_experiment_info_func(){
   _monitor_experiment_info_func=( "$@" )
 }
 
+_monitor_login(){
+
+  local output code
+
+  rm -f "$_monitor_cookie_file"
+
+  output=$(curl -sS -w '\n%{http_code}\n' -XPOST https://${_monitor_eshost}/auth/log_in -H 'Content-type: application/json' \
+                -c "$_monitor_cookie_file" -b "$_monitor_cookie_file" -d '{"email": "'"$_monitor_proxy_email"'", "password": "'"$_monitor_proxy_password"'" }')
+
+  code=$( echo "$output" | tail -n1)
+
+  if [ "$code" = "200" ]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+_monitor_send(){
+
+  local url=$1
+  local json=$2
+  local log=$3
+
+  local total out code
+
+  if [ $_monitor_update_es -eq 0 ]; then
+    return
+  fi
+
+  total=$(curl -sS -w '\n%{http_code}\n' -c "$_monitor_cookie_file" -b "$_monitor_cookie_file" -XPOST "$url" -H 'Content-Type: application/json' --data-binary "$json")
+  out=$(echo "$total" | head -n -1)
+  code=$(echo "$total" | tail -n 1)
+
+  if [ "$code" = "401" ]; then
+    _monitor_login
+
+    if [ $? -eq 0 ]; then
+      total=$(curl -sS -w '\n%{http_code}\n' -c "$_monitor_cookie_file" -b "$_monitor_cookie_file" -XPOST "$url" -H 'Content-Type: application/json' --data-binary "$json")
```

Comment:
Seems like `curl -sS -w '\n%{http_code}\n' -c "$_monitor_cookie_file" -b "$_monitor_cookie_file" -XPOST "$url" -H 'Content-Type: application/json'` is being repeated in a few places, could be made into a function.